lms/services/application_instance.py

```diff
@@ -18,7 +18,7 @@ def __init__(self, db, request, aes_service):
         self._request = request
         self._aes_service = aes_service
 
-    @lru_cache
+    @lru_cache(maxsize=1)
```

Comment:
I always thought `lru_cache` without a value was iffy. It turns out it causes `self` to be included in the cache, and it's never evicted, preventing garbage collection. This might reduce some of our memory leaks?

In the case where the method takes no arguments I've set the `maxsize` to 1. In the cases where it does take arguments I've used 128 which is the default listed in `lru_cache`.